lib/maths/CBoostedTreeFactory.cc

```diff
@@ -1017,5 +1022,11 @@ void CBoostedTreeFactory::noopRecordProgress(double) {
 
 void CBoostedTreeFactory::noopRecordMemoryUsage(std::int64_t) {
 }
+
+CBoostedTreeFactory& CBoostedTreeFactory::topShapValues(std::size_t topShapValues) {
+    m_TopShapValues = topShapValues;
```

Comment:
Do we need this variable on the factory as well: seems redundant?

Reply:
I need it to estimate memory usage with resized data frame.